### PR TITLE
Small perf infrastructure changes

### PR DESCRIPTION
**Pull request by jorive**
- Move xUnit Perf Api output files to a folder `Logs` under sandbox
- Made necessary changes to `perf.groovy` to archive artifacts
- Added a switch -outputdir to the `run-xunit-perf*` scripts
- Enable running all CoreClr scenarios (*JitBench* & *IlLink*) despite failure

### CODE REVIEW

**Comment by jorive** [comment]
@dotnet-bot test ci please

File: perf.groovy [hunk 1]
```diff
@@ -309,8 +308,8 @@ def static getFullPerfJobName(def project, def os, def isPR) {
 
         // Save machinedata.json to /artifact/bin/ Jenkins dir
         def archiveSettings = new ArchivalSettings()
-        archiveSettings.addFiles('Perf-*.log')
-        archiveSettings.addFiles('Perf-*.xml')
+        archiveSettings.addFiles('${outputDirectory}/Perf-*.log')
+        archiveSettings.addFiles('${outputDirectory}/Perf-*.xml')
         archiveSettings.addFiles('machinedata.json')
```
Comment (dpodder) [inline perf.groovy]: Are these still generated at the root? Should we move them into sandbox/Logs also?

**Comment by jorive** [comment]
@dotnet-bot test ci please

File: tests/scripts/run-xunit-perf.sh [hunk 2]
```diff
@@ -1,5 +1,7 @@
 #!/usr/bin/env bash
 
+dp0=$( cd -P "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
```
Comment (dpodder) [inline tests/scripts/run-xunit-perf.sh]: I think we should really rename this to script_dir instead. 😝

File: tests/src/Common/PerfHarness/PerfHarness.csproj [hunk 3]
```diff
@@ -1,12 +1,17 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
+  <Import Project="..\..\..\..\dependencies.props" />
```
Comment (dpodder) [inline tests/src/Common/PerfHarness/PerfHarness.csproj]: You should use

```xml
<Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dependencies.props))\dependencies.props" />
```

File: perf.groovy (hunk 1 above)
Comment (jorive) [inline perf.groovy]: No, this is the new location to get the output files to be archive (used to be generated in sandbox and copied to the CORECLR_ROOT).

File: tests/src/Common/PerfHarness/PerfHarness.csproj (hunk 3 above)
Comment (jorive) [inline tests/src/Common/PerfHarness/PerfHarness.csproj]: I can try, but I do not think the "new" csproj fully understand the "old" msbuild ways.

File: tests/scripts/run-xunit-perf.sh (hunk 2 above)
Comment (jorive) [inline tests/scripts/run-xunit-perf.sh]: I might leave it for nostalgic reasons :p

File: perf.groovy [hunk 4]
```diff
@@ -562,20 +560,29 @@ parallel(
 
                     batchFile("tests\\runtest.cmd ${configuration} ${architecture} GenerateLayoutOnly")
 
+                    def runXUnitPerfCommonArgs = "-arch ${arch} -configuration ${configuration} -generateBenchviewData \"%WORKSPACE%\\Microsoft.Benchview.JSONFormat\\tools\" ${uploadString} -runtype ${runType} -outputdir \"${outputDirectory}\" -scenarioTest"
+                    def failedOutputLogFilename = "run-xunit-perf-scenario.log"
+
+                    // Using a centinel file to
+                    batchFile("if exist \"${failedOutputLogFilename}\" del /q /f \"${failedOutputLogFilename}\"")
+                    batchFile("if exist \"${failedOutputLogFilename}\" (echo [ERROR] Failed to previously created \"${failedOutputLogFilename}\" file.& exit /b 1)")
+
```
Comment (AndyAyersMS) [inline perf.groovy]: Couple of typos: `centinel` --> `sentinel` and `failed to **delete** previously...`

File: tests/scripts/run-xunit-perf.cmd [hunk 5]
```diff
@@ -2,13 +2,14 @@
...
 
-@echo off
-@if defined _echo echo on
+@if not defined _echo echo off
 
 setlocal ENABLEDELAYEDEXPANSION
   set ERRORLEVEL=
   set BENCHVIEW_RUN_TYPE=local
   set CORECLR_REPO=%CD%
+  set LV_SANDBOX_DIR=%CORECLR_REPO%\sandbox
```
Comment (AndyAyersMS) [inline tests/scripts/run-xunit-perf.cmd]: Does it make more sense to have the defai;t sandbox live under `bin` with similar derived / built stuff?

Comment (jorive) [inline tests/scripts/run-xunit-perf.cmd]: I like it! I'll make the change!

**Comment by jorive** [comment]
@dotnet-bot test ci please

**Comment by jorive** [comment]
@dotnet-bot test ci please

**Comment by jorive** [comment]
@dotnet-bot test ci please

**Comment by jorive** [comment]
@dotnet-bot test ci please

**Comment by jorive** [comment]
@dotnet-bot test ci please

**Comment by jorive** [comment]
@dotnet-bot test ci please

**Pull request merged**